consensus/istanbul/backend/snapshot.go

```diff
@@ -63,13 +64,19 @@ func loadSnapshot(epoch uint64, db ethdb.Database, hash common.Hash) (*Snapshot,
 	if err := json.Unmarshal(blob, snap); err != nil {
 		return nil, err
 	}
+
+	if !snap.ValSet.HasBLSKeyCache() {
```

Comment:
why fail? why not just cache it here? cc @kobigurk

Reply:
Caching + storing makes sense to me...

Reply:
It fails so that it's cached in the DB.